1.74.0-beta1: Android build is broken

When building with NDK r20 for `armeabi-v7a` with `android-19` backward compatibility I get the following error:

```
libs/nowide/src/filebuf.cpp:62:30: error: use of undeclared identifier 'ftello'; did you mean 'ftell'?
            const auto pos = BOOST_NOWIDE_FTELL(file);
                             ^~~~~~~~~~~~~~~~~~
                             ftell
libs/nowide/src/filebuf.cpp:29:28: note: expanded from macro 'BOOST_NOWIDE_FTELL'
#define BOOST_NOWIDE_FTELL ftello
                           ^
libs/nowide/src/filebuf.cpp:60:24: note: 'ftell' declared here
        std::streampos ftell(FILE* file)
                       ^
libs/nowide/src/filebuf.cpp:76:20: error: use of undeclared identifier 'fseeko'; did you mean 'fseek'?
            return BOOST_NOWIDE_FSEEK(file, static_cast<BOOST_NOWIDE_OFF_T>(offset), origin);
                   ^~~~~~~~~~~~~~~~~~
                   fseek
libs/nowide/src/filebuf.cpp:30:28: note: expanded from macro 'BOOST_NOWIDE_FSEEK'
#define BOOST_NOWIDE_FSEEK fseeko
                           ^
libs/nowide/src/filebuf.cpp:70:13: note: 'fseek' declared here
        int fseek(FILE* file, std::streamoff offset, int origin)
            ^
libs/nowide/src/filebuf.cpp:48:13: error: static_assert failed "Mixed sign comparison can lead to problems below"
            static_assert(std::is_signed<T>::value == std::is_signed<U>::value,
            ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libs/nowide/src/filebuf.cpp:57:20: note: in instantiation of function template specialization 'boost::nowide::detail::is_in_range<long long, std::__ndk1::fpos<mbstate_t> >' requested here
            return is_in_range<T>(value) ? static_cast<T>(value) : T(-1);
                   ^
libs/nowide/src/filebuf.cpp:67:20: note: in instantiation of function template specialization 'boost::nowide::detail::cast_if_valid_or_minus_one<long long, std::__ndk1::fpos<mbstate_t> >' requested here
            return cast_if_valid_or_minus_one<std::streamoff>(pos);
                   ^
3 errors generated.
```

In the Android `stdio.h` the `ftello` and friends are restricted to `android-24` and newer:
```
#if __ANDROID_API__ >= 24
int fgetpos(FILE* __fp, fpos_t* __pos) __RENAME(fgetpos64) __INTRODUCED_IN(24);
int fsetpos(FILE* __fp, const fpos_t* __pos) __RENAME(fsetpos64) __INTRODUCED_IN(24);
int fseeko(FILE* __fp, off_t __offset, int __whence) __RENAME(fseeko64) __INTRODUCED_IN(24);
off_t ftello(FILE* __fp) __RENAME(ftello64) __INTRODUCED_IN(24);
#endif /* __ANDROID_API__ >= 24 */
```

Somehow the LFS check return true:
```
- Has Large File Support   : yes
```

Should I prepare instructions how to reproduce?

If I make the check define also `_FILE_OFFSET_BITS` it fails like the source code:
```
#define _LARGEFILE_SOURCE
#ifndef _FILE_OFFSET_BITS
#define _FILE_OFFSET_BITS 64
#endif
```

```
~/Android/android-ndk-r20/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi19-clang++ -c check_lfs_support.cpp 
check_lfs_support.cpp:25:11: error: use of undeclared identifier 'fseeko'; did you mean 'fseek'?
    (void)fseeko(f, off_t(0), SEEK_CUR);
          ^~~~~~
          fseek
/home/gregorj/Android/android-ndk-r20/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/stdio.h:175:5: note: 'fseek' declared here
int fseek(FILE* __fp, long __offset, int __whence);
    ^
check_lfs_support.cpp:26:11: error: use of undeclared identifier 'ftello'; did you mean 'ftell'?
    (void)ftello(f);
          ^~~~~~
          ftell
/home/gregorj/Android/android-ndk-r20/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/stdio.h:176:6: note: 'ftell' declared here
long ftell(FILE* __fp);
     ^
2 errors generated.
```

--- a/build/Jamfile.v2
+++ b/build/Jamfile.v2
@@ -16,7 +16,7 @@ local requirements =
 
 obj cxx11_moveable_fstreams : ../test/check_movable_fstreams.cpp ;
 explicit cxx11_moveable_fstreams ;
-obj lfs_support : ../test/check_lfs_support.cpp ;
+exe lfs_support : ../test/check_lfs_support.cpp ;
 explicit lfs_support ;
 
 project boost/nowide

--- a/test/check_lfs_support.cpp
+++ b/test/check_lfs_support.cpp
@@ -6,6 +6,10 @@
 //
 
 #define _LARGEFILE_SOURCE
+#ifndef _FILE_OFFSET_BITS
+#define _FILE_OFFSET_BITS 64
+#endif
+
 #include <stdio.h>
 
 void check(FILE* f)

